tests/tensor_search/test_search.py

```diff
@@ -574,27 +581,6 @@ def test_filtering_bad_syntax(self):
         except InvalidArgError:
             pass
 
-    def test_set_device(self):
```

Comment:
This is a good test to have, but can be refactored so that it overrides the BEST_DEVICE env vr